packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -158,11 +160,14 @@ class _MaterialTextSelectionControls extends TextSelectionControls {
     final double x = (endTextSelectionPoint == null)
         ? startTextSelectionPoint.point.dx
         : (startTextSelectionPoint.point.dx + endTextSelectionPoint.point.dx) / 2.0;
-    final double availableHeight
-        = globalEditableRegion.top - MediaQuery.of(context).padding.top - _kToolbarScreenPadding;
-    final double y = (availableHeight < _kToolbarHeight)
-        ? startTextSelectionPoint.point.dy + globalEditableRegion.height + _kToolbarHeight + _kToolbarScreenPadding
-        : startTextSelectionPoint.point.dy - globalEditableRegion.height;
+    final double toolbarHeightNeeded = MediaQuery.of(context).padding.top
+      + _kToolbarScreenPadding
+      + _kToolbarHeight;
```

Comment:
Is there supposed to be a non-zero content padding (between the paragraph and the bottom of the toolbar)?

Reply:
It doesn't seem so looking at the code, but looking at native apps it seems like there should be.  I'll add in a `_kToolbarContentDistance` like in iOS.  It may break golden/scuba tests.

Reply:
After:

<img width="354" alt="Screen Shot 2019-07-30 at 8 32 30 AM" src="https://user-images.githubusercontent.com/389558/62143323-aa1e3d80-b2a4-11e9-8ba0-005c2dce1893.png">
<img width="316" alt="Screen Shot 2019-07-30 at 8 32 56 AM" src="https://user-images.githubusercontent.com/389558/62143324-aa1e3d80-b2a4-11e9-83f7-8692b4fbdebd.png">